R/limesInt2Ext.R

```diff
@@ -1,86 +1,96 @@
-  posint_tmp <- posint_tmp[pos_tmp]
-  posext_tmp <- posext_tmp[pos_tmp]
-  #Concatenate the names of the intensive and extensive variables
-  int_tmp <- paste0(mapping_vars$LIMES[posint_tmp]," (",mapping_vars$UnitLIMES[posint_tmp],")")
-  ext_tmp <- paste0(mapping_vars$LIMES[posext_tmp]," (",mapping_vars$UnitLIMES[posext_tmp],")")
-  
+
+  ##Find the position of the desired variables in mapping variable list to add the units
+  #posint_tmp <- match(mapping_int2ext$int,mapping_vars$LIMES)
+  #posext_tmp <- match(mapping_int2ext$ext,mapping_vars$LIMES)
+  ##Make sure that only variables where corresponding "weighter" are kept
+  #pos_tmp <- intersect(which(!is.na(posint_tmp)),which(!is.na(posext_tmp)))
+  #posint_tmp <- posint_tmp[pos_tmp]
+  #posext_tmp <- posext_tmp[pos_tmp]
+  ##Concatenate the names of the intensive and extensive variables
+  #int_tmp <- paste0(mapping_vars$LIMES[posint_tmp]," (",mapping_vars$UnitLIMES[posint_tmp],")")
+  #ext_tmp <- paste0(mapping_vars$LIMES[posext_tmp]," (",mapping_vars$UnitLIMES[posext_tmp],")")
+  ##Check if the variables (intensive and corresponding extensive) were previously calculated (output)
+  #pos_int <- match(int_tmp, var_names)
+  #pos_ext <- match(ext_tmp, var_names)
+  #pos_tmp2 <- intersect(which(!is.na(pos_int)),which(!is.na(pos_ext)))
+  #int_tmp2 <- int_tmp[pos_tmp2]
+  #ext_tmp2 <- ext_tmp[pos_tmp2]
+
+  #Keep only names in var_names, i.e., delete units
+  var_names_tmp <- gsub(" [(].*", "", var_names)
+
   #Check if the variables (intensive and corresponding extensive) were previously calculated (output)
-  pos_int <- match(int_tmp, var_names)
-  pos_ext <- match(ext_tmp, var_names)
+  pos_int <- match(mapping_int2ext$int, var_names_tmp)
+  pos_ext <- match(mapping_int2ext$ext, var_names_tmp)
+
+  #Make sure only the intensive variables for which the corresponding variable is available, are weighted (avoid errors)
   pos_tmp2 <- intersect(which(!is.na(pos_int)),which(!is.na(pos_ext)))
-  int_tmp2 <- int_tmp[pos_tmp2]
-  ext_tmp2 <- ext_tmp[pos_tmp2]
-  
+  int_tmp2 <- mapping_int2ext$int[pos_tmp2]
+  ext_tmp2 <- mapping_int2ext$ext[pos_tmp2]
+
+  #Retrieve variables with their corresponding units
+  int <- var_names[match(int_tmp2, var_names_tmp)]
+  ext <- var_names[match(ext_tmp2, var_names_tmp)]
+
+  #Allocate very low values for extensive variables where values are 0
+  var[, , ext] <- pmax(var[, , ext], 1e-20)
+
   #Create a magpie object for each intensive variable
   regionSubsetList <- list("GLO" = getItems(var, dim = 1),
                            "EU28" = setdiff(getItems(var, dim = 1), c("CHE","NOR","BAL")),
                            "EUETS" = setdiff(getItems(var, dim = 1), c("CHE","BAL")))
-  
-  tmp_RegAgg <- new.magpie(cells_and_regions = names(regionSubsetList),years = getYears(var),names = int_tmp2,fill=0)
-  
+
+  tmp_RegAgg <- new.magpie(cells_and_regions = names(regionSubsetList), years = getYears(var), names = int, fill=0)

```

Comment:
For more clarity, I would suggest:

```suggestion
 tmp_RegAgg <- new.magpie(cells_and_regions = names(regionSubsetList), 
                                                years = getYears(var),
                                                names = int,
                                                fill=0)

```